### PR TITLE
KEYCLOAK 8735 - updates to upgrade guide

### PR DESCRIPTION
**Pull request by matthewhelmke**
These changes are for the upcoming 7.3 release of RH-SSO. I have based my changes here on the changes noted in the Keycloak docs for 4.0.0 through 4.6.0.

I _think_ I have captured everything, but would appreciate some help from engineering to review and check all the details. Also, if I missed anything I don't know what it is and would love a nudge in the right direction.

### CODE REVIEW

**Comment by matthewhelmke** [comment]
I see this picked up some commits from @iankko . I do not know why. I did a fetch/rebase before making my changes which seemed to go normally. Then I turned around and pushed this fairly quickly. Odd.

**Comment by iankko** [comment]
@matthewhelmke AFAICT the possible reason how this could happen is before you have pushed the KEYCLOAK-8129 changes to your master per this comment:
  https://github.com/keycloak/keycloak-documentation/pull/455#issuecomment-438245158

and if you then created KEYCLOAK-8735 branch from your master, the previous KEYCLOAK-8129 commits were still there.

In any case, you can drop the two KEYCLOAK-8129 related commits (updated my PR, so they will arrive within my KEYCLOAK-8129  change), and keep in your PR just the 3rd commit.

**Comment by matthewhelmke** [comment]
@iankko that makes sense. I had forgotten that I had done that when I saw your PR update. Thank you.

I'll research how to drop the unrelated commits from this PR.

**Comment by matthewhelmke** [comment]
Got it figured out. Thanks, again, @iankko.

@stianst I believe this is ready for review, with the original comment still in force.

File: upgrading/topics/rhsso/changes-73.adoc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+endpoints for authorization and obtaining the user profile. From November 2018 onwards, Microsoft is removing support
+for the Live SDK API in favor of the new Microsoft Graph API. The {project_name} identity provider has been updated
+to use the new endpoints so if this integration is in use make sure you upgrade to {project_name} version 4.6.0 or later.
+
+Legacy client applications registered under "Live SDK applications" won't work with the Microsoft Graph endpoints
+due to changes in the id format of the applications. If you run into an error saying that the application identifier
+was not found in the directory, you will have to register the client application again in the
+https://account.live.com/developers/applications/create[Microsoft Application Registration] portal to obtain a new application id.
+
+=== Upgrade to Wildfly 14
```
Comment (stianst) [inline upgrading/topics/rhsso/changes-73.adoc]: This needs to be changed to EAP 7.2. Same with references below to WF14.

File: upgrading/topics/rhsso/changes-73.adoc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+endif::[]
+
+=== Hostname configuration
+
+In previous versions it was recommended to use a filter to specify permitted hostnames. It is now possible to
+set a fixed hostname which makes it easier to make sure the valid hostname is used and also allows internal
+applications to invoke {project_name} through an alternative URL, for example an internal IP address. It is
+recommended that you switch to this approach in production.
+
+=== Client Templates changed to Client Scopes
```
Comment (stianst) [inline upgrading/topics/rhsso/changes-73.adoc]: Should combine all sections for client scopes into a single section

**Review by stianst — CHANGES_REQUESTED** [review]
Items should be listed in priority order:

* Authorization Services
* Client Scope
* Upgrade to EAP 7.2 
* Hostname configuration
* JavaScript adapter promise
* Microsoft Identity Provider

**Comment by matthewhelmke** [comment]
I will work on this one first thing tomorrow morning.

**Comment by matthewhelmke** [comment]
@stianst The new commit includes all requested changes.

**Review by stianst — APPROVED** [review]

**Pull request merged**